pycbc/inject.py

```diff
@@ -169,12 +172,44 @@ def apply(self, strain, detector_name, f_lower=None, distance_scale=1,
             signal_lal = signal.lal()
             add_injection(lalstrain, signal_lal, None)
             injection_parameters.append(inj)                            
-            
+            if strain.injcutter is not None and \
```

Comment:
I don't think I like having the special injections made within the standard injection path. The overall structure is very messy. Can this be modularized into a separate injection method to be called from the injection rejection code directly?